web/packages/teleterm/src/services/tshd/types.ts

```diff
@@ -403,6 +404,10 @@ export type CreateAccessRequestParams = {
   roles: string[];
   suggestedReviewers: string[];
   resourceIds: { kind: ResourceKind; clusterName: string; id: string }[];
+  assumeStartTime?: Date;
+  dryRun?: boolean;
+  maxDuration?: Date;
+  requestTtl?: Date;
```

Comment:
We should use timestamps instead of dates. We plan to keep the types from `tshd/types` and `apiService` as similar as possible. The conversion should happen directly in the callsite.

Reply:
Cool, thanks. So the type that I posted does exactly that since the timestamp fields get "inherited" from `apiService.CreateAccessRequestRequest` directly.

Reply:
@ravicious @gzdunek i'm getting a type conflict because previously `ResourceId` contained a field `id` and was assigned this to a field called `name`:

```
      resourceIds: params.resourceIds.map(({ id, clusterName, kind }) =>
          ResourceID.create({
            name: id,
            clusterName,
            kind,
          })
        ),
```

does `id` == `subResourceName`?

Reply:
Actually, couldn't the type look like this?
```ts
export type CreateAccessRequestParams =
  apiService.CreateAccessRequestRequest & {
    rootClusterUri: uri.RootClusterUri;
  };
```

I think it doesn't hurt us to provide `false` for `dryRun` and `''` as `subResourceName` in the callsite.

Reply:
i made the change [here](https://github.com/gravitational/teleport/pull/39050/commits/064a3522347144ce2e255abe26e926f84774e41e) with the enterprise change. i ran `yarn type-check` and didn't run into errors. any concerns @ravicious ?